lib/logstash/shutdown_controller.rb

```diff
@@ -0,0 +1,98 @@
+# encoding: utf-8
+require 'pp'
+
+module LogStash
+  class ShutdownController
+
+    CHECK_EVERY = 1 # second
+    REPORT_EVERY = 5 # checks
+    ABORT_AFTER = 3 # stalled reports
+    REPORTS = []
+
+    def self.force_shutdown=(boolean)
+      @force_shutdown = boolean
+    end
+
+    def self.force_shutdown?
+      @force_shutdown
+    end
+
+    def logger=(logger)
+      @logger = logger
+    end
+
+    def logger
+      @logger ||= Cabin::Channel.get(LogStash)
+    end
+
+    def initialize(pipeline)
+      @pipeline = pipeline
+    end
+
+    def start(cycle_period=CHECK_EVERY, report_every=REPORT_EVERY, abort_threshold=ABORT_AFTER)
+      @thread ||= Thread.new do
```

Comment:
I am looking at the calling code and I don't see where `start` could be called on the same instance?
I don't think we need to lazy instantiate the @thread variable